library/ssl_msg.c

```diff
@@ -2984,7 +2984,8 @@ int mbedtls_ssl_write_handshake_msg_ext( mbedtls_ssl_context *ssl,
         if( ( ret = mbedtls_ssl_write_record( ssl, SSL_FORCE_FLUSH ) ) != 0 )
         {
             MBEDTLS_SSL_DEBUG_RET( 1, "ssl_write_record", ret );
-            return( ret );
+            if(ret != MBEDTLS_ERR_SSL_WANT_WRITE)
+                return( ret );
```

Comment:
Trying to understand, if we do not return here, we would return on line 2994 with 0, right? Would that confuse the caller of this function to wrongfully assume the write has been successful?

Reply:
Yeah, we may need propagate `MBEDTLS_ERR_SSL_WANT_WRITE` to upper level.

Reply:
I agree, we can't ignore the error. @yuhaoth What was the motivation behind this change?

Reply:
As @lhuang04  said , propagate it  to upper level .